lib/Analysis/NoiseAnalysis/BFV/NoiseAnalysis.cpp

```diff
@@ -55,10 +55,15 @@ LogicalResult NoiseAnalysis<NoiseModel>::visitOperation(
   };
 
   auto propagate = [&](Value value, NoiseState noise) {
-    LLVM_DEBUG(llvm::dbgs()
-               << "Propagating "
-               << noiseModel.toLogBoundString(getLocalParam(value), noise)
-               << " to " << value << "\n");
+    [[maybe_unused]] auto doubleToString = [](double d) {
+      std::stringstream stream;
+      stream << std::fixed << std::setprecision(2) << d;
+      return stream.str();
+    };
+    LLVM_DEBUG(llvm::dbgs() << "Propagating "
+                            << doubleToString(noiseModel.toLogBound(
+                                   getLocalParam(value), noise))
+                            << " to " << value << "\n");
```

Comment:
Rather than define a `maybe_unused` function, put the function in a utils class somewhere (or in Noise.h?) to avoid redefining it where you use it. Or just use the block form of `LLVM_DEBUG` to skip the function definition altogether.

```suggestion
    LLVM_DEBUG({
      std::stringstream stream;
      stream << std::fixed << std::setprecision(2)
             << noiseModel.toLogBound(
                  getLocalParam(value), noise);
      llvm::dbgs() << "Propagating "
                            << stream.str()
                            << " to " << value << "\n";
    });
```

Reply:
Turned out `RLWEParams.cpp` also needs this function and I put it in `lib/Utils/Utils.h`